support-frontend/assets/pages/contributions-landing/contributionsLandingActions.js

```diff
@@ -251,6 +261,31 @@ const setUserTypeFromIdentityResponse =
         ({ type: 'SET_USER_TYPE_FROM_IDENTITY_RESPONSE', userTypeFromIdentityResponse })));
     };
 
+// We defer loading 3rd party payment SDKs until the user selects one, or one is selected by default
+const loadPaymentSdkIfNecessary = (
```

Comment:
Would it be cleaner to split this into 2 functions, and have the 2 callees listen to the relevant 'begun-loading' variable and select the right function name, rather than passing them through?

`const loadPayPalSdkIfNecessary = (contributionType) => (dispatch: Function): void`
`loadAmazonPaySdkIfNecessary = (countryGroupId, isTestUser), => (dispatch: Function): void`

Reply:
yes, but then each callee has to `switch` on the `paymentMethod`, which seemed worse to me. Which do you prefer?

Reply:
There are only 2 callees at the moment, so I vote for separate functions rather than argument expansion. Least power principle applies to arguments sets as well as argument types.

Reply:
👍